### PR TITLE
feat: support metadrive env

### PR DESCRIPTION
**Pull request by Gaiejj**
# Description

**This PR is already complete in terms of implementation accuracy. We will merge it shortly after improving the code style and documentation.**

This PR supports `SafeMetaDrive` from [MetaDrive](https://github.com/metadriverse/metadrive)[1].

[1]: [Metadrive: Composing diverse driving scenarios for generalizable reinforcement learning](https://arxiv.org/pdf/2109.12674.pdf)

Autonomous driving is an important application of reinforcement learning algorithms, and safety is particularly crucial. Safety constraints for autonomous driving tasks are an important research direction in the SafeRL field. We have implemented the embedding of SafeMetaDrive based on our customized environment interface (more details in #310 ) and presented the SafeRL baseline algorithms involved in the MetaDrive paper as follows, hoping to inspire research on related algorithms.

image

## Motivation and Context

Related issues can be found at #262 

- [x] I have raised an issue to propose this change ([required](https://github.com/PKU-Alignment/omnisafe/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/PKU-Alignment/omnisafe/blob/HEAD/CONTRIBUTING.md) guide. (**required**)
- [x] My change requires a change to the documentation.
- [x] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [x] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format`. (**required**)
- [x] I have checked the code using `make lint`. (**required**)
- [x] I have ensured `make test` pass. (**required**)

### CODE REVIEW

File: omnisafe/envs/meta_drive_env.py [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+    ) -> None:
+        """Initialize an instance of :class:`SafetyGymnasiumEnv`."""
+        super().__init__(env_id)
+        self._num_envs = num_envs
+        self._device = torch.device(device)
+
+        if META_DRIVE_AVAILABLE:
+            self._env = SafeMetaDriveEnv(config=kwargs.get('meta_drive_config'))
+        else:
+            raise ImportError('Please install MetaDrive to use Safe SafeMetaDrive!')
```
Comment (muchvo) [inline omnisafe/envs/meta_drive_env.py:86]:
```suggestion
            raise ImportError('Please install MetaDrive to use SafeMetaDrive!')
```

Comment (muchvo) [reply omnisafe/envs/meta_drive_env.py:86]: Can we further recommend the command, such as `pip install metadrive`?

Comment (Gaiejj) [reply omnisafe/envs/meta_drive_env.py]: It is a pretty nice idea! I have made a revision.

**Review by muchvo — APPROVED** [review]
LGTM.

**Pull request merged**